iceoryx2/src/port/publisher.rs

```diff
@@ -982,11 +983,25 @@ impl<Service: service::Service, Payload: Debug, UserHeader: Debug>
             ))
         };
 
+        let payload_size = self
+            .data_segment
+            .subscriber_connections
+            .static_config
+            .message_type_details
+            .payload
+            .size;
```

Comment:
At some point we might need to cache this value within the publisher. I would guess this will lead to quit a few cache misses.